client/src/pages/EditProfilePage/EditProfilePage.tsx

```diff
@@ -21,32 +22,167 @@ const EditProfilePage = () => {
+      },
+    },
+    socialMediaLinks: {
+      twitter: "",
+      blog: "",
+    },
+    availabilityForNetworking: false,
   });
 
+ 
+
   const [file, setFile] = useState<File | null>(null);
 
   useEffect(() => {
     if (userID) dispatch(fetchUserProfile(userID as string));
   }, [dispatch]);
+
   useEffect(() => {
     if (profile) {
       setFormData({
+        cohort: profile.cohort || "",
         email: profile.email || "",
+        firstName: profile.firstName || "",
+        linkedInProfile: profile.linkedInProfile || "",
+        gitHubProfile: profile.gitHubProfile || "",
+        nickName: profile.nickName || "",
         personalBio: profile.personalBio || "",
+        skills: profile.skills || [],
+        specializations: profile.specializations || [],
+        careerInformation: {
+          currentPosition: {
+            title: profile.careerInformation?.currentPosition?.title || "",
+            company: profile.careerInformation?.currentPosition?.company || "",
+          },
+        },
+        socialMediaLinks: {
+          twitter: profile.socialMediaLinks?.twitter || "",
+          blog: profile.socialMediaLinks?.blog || "",
+        },
+        availabilityForNetworking: profile.availabilityForNetworking || false,
       });
     }
   }, [profile]);
 
-  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+  const handleChange = (
+    e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+  ) => {
     const { name, value } = e.target;
     setFormData((prevFormData) => ({
       ...prevFormData,
       [name]: value,
     }));
   };
+  const handleCompanyChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.target;
+    setFormData((prevData) => ({
+      ...prevData,
+      careerInformation: {
+        ...prevData.careerInformation,
+        currentPosition: {
+          ...prevData.careerInformation.currentPosition,
+          company: value,
+        },
+      },
+    }));
+  };
+  
+ const [skillInput, setSkillInput] = useState("");
+
+  const handleSkillChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setSkillInput(e.target.value);
+  };
+
+  const handleSpecialization = (skill: string) => {
+    setFormData((prevData) => {
+      const isAlreadySpecialized = prevData.specializations.includes(skill);
+      return isAlreadySpecialized
+        ? prevData
+        : {
+            ...prevData,
+            specializations: [...prevData.specializations, skill],
+          };
+    });
+  };
+
+  const handleSkillKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === "Enter" && skillInput.trim() !== "") {
+      e.preventDefault();
+      if (!formData.skills.includes(skillInput.trim())) {
+        setFormData((prevData) => ({
+          ...prevData,
+          skills: [...prevData.skills, skillInput.trim()],
+        }));
+      }
+      setSkillInput("");
+    }
+  };
+
+  const handleSkillRemove = (skill: string, specialty: boolean) => {
```

Comment:
More of a possible Profile model refactor question but should we allow specializations that aren't skills?

Reply:
Good question! I think we should definitely allow this in the future. However, I discussed with @brok3turtl3 and we are going to leave this for another ticket.